### PR TITLE
Added Scene Quality option, making sure all codecs are detected

### PR DESCRIPTION
**Pull request by medariox**
Won't use regex for 10 elements. Please close if not okay with that.

### CODE REVIEW

**Comment by SiCKRAGEB0T** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your report has all the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure you send a PR against the DEVELOP branch
- [x] Don't send big changes in one go. Split up big PRs into multiple smaller (easier manageble) PRs

Please make sure you also read [contribution guide](https://github.com/SiCKRAGETV/SickRage/blob/master/contributing.md) and followed all the steps.

Thanks!

File: sickbeard/tv.py [hunk 1]
```diff
@@ -2177,7 +2200,7 @@ def release_group(show, name):
             '%AB': '%(#)03d' % {'#': self.absolute_number},
             '%XAB': '%(#)03d' % {'#': self.scene_absolute_number},
             '%RN': release_name(self.release_name),
-            '%RG': rel_grp[relgrp],
+            '%RG': rel_grp[relgrp].upper(),
```
Comment (miigotu) [inline sickbeard/tv.py]: ????

Comment (medariox) [inline sickbeard/tv.py]: %RG would just output the group name as it is in the file name (doesn't matter if capitalized or not), but the capitlized %RG should output the capitalized group name (like the other parameters). Now the user has the option to output it capitalized %RG or %rg lowercase.

**Comment by miigotu** [comment]
release group is supposed to stay as it came. `IMMERSE` is not the same as `iMMERSE`

**Comment by fernandog** [comment]
agree with miigotu

**Comment by medariox** [comment]
If all torrents were perfectly tagged, there wouldn't be any need for all of this. Sadly some torrents come as `iMMERSE` other as `IMMERSE` and some others as `immerse`. So what do you suggest? Maybe add an option to keep the "orginal" (aka as it comes from the file name)?

**Comment by miigotu** [comment]
Those are all different people uploading them. Real ones, untampered with, are iMMERSE.

The distinction needs to be left there.

**Comment by miigotu** [comment]
Kinda strange you started this because you didnt want to lose the codec information but are ok with putting incorrect codec info or incorrect group info.

**Comment by medariox** [comment]
Hm? What are you talking about again?
If someone uploads a release like this:
`my.tv.show.112.hdtv-immerse`
SR should find a way to correctly parse the above file name to a proper release name.
e.g. `%S.N.S%0SE%0E.%RT.%SQ.N-%RG` would be like a proper scene release name, except for the group name, which will stay lowercase, breaking the whole point of the name pattern:
`My.Tv.Show.S01E12.HDTV.x264-immerse`
Since any other patter can be either capitalized (%RG) or lowercase (%rg), I don't see why it shouldn't be possible for the group info. It just doesn't make any sense to not add the option.
If for you having `IMMERSE` instead of `iMMERSE` is having "incorrect group info", I'm perfectly fine with that. The codec is only "incorrectly" added when there is no way of telling the real codec of the releases (aka it is not in the file name), e.g. `my.tv.show.112.hdtv-immerse` - why don't you understand that?

**Comment by fernandog** [comment]
Im all for give users options. But default should be the default

**Comment by miigotu** [comment]
The point of the name pattern is to PRESERVE information.
If the codec is not in the file name to begin with is should not be added to the result at all.

You cannot force naming to be absolutely 100% conforming or normalized by falsifying information and blatantly lying in the output. Making assumptions or changing the information that is already there does not fit in the equation.

Normalizing show names, season and episode numbers, and even codec and source is OK because they have an indisputably CORRECT format.

For release groups, unless you are going to add a map that includes EVERY release group in the world so that you can format them as is their proper name you cannot change them. These are PROPER NOUNS mixed with the complication of a trademark.

Have you not thought about why it wasnt uppercased or lowercased in the code to begin with? This is why.

**Comment by fernandog** [comment]
The original issue isnt the SR renaming was not considering code even if it exist in file name?

**Comment by miigotu** [comment]
Yes the original issue was codec missing. Ok add a way to get the codec (accurately, and normalize it correctly). But dont add false information or go changing other things in the same PR that are the way they are for a reason, especially when they are unrelated changes.

**Comment by medariox** [comment]
@fernandog @miigotu 
I can add an option to use the capitalized group name (%CRG instead of %RG for example), but I won't change the adding of x264 when there is no codec in the file name, because I get many, many torrents that are like that and Scene Quality pattern will try to make it scene standard, even when some crazy uploader decides that they can strip information from the file name. If you are not okay with that, please close the PR.

**Comment by fernandog** [comment]
Cant you PR the fix only for now??

**Comment by medariox** [comment]
@fernandog 
It has no point as miigotu won't take it and I'm tired of adding features nobody wants, so I will just keep it private for myself.

**Comment by miigotu** [comment]
@fernandog he wants to add x264 blindly to any file we cannot know the codec of.

AKA:
(release name)
some.show.s01e01.hdtv.wmv
becomes:
Some.Show.S01E01.SDTV.x264.wmv

That is the most unacceptable thing about this PR.

**Comment by fernandog** [comment]
@medariox why don't you check the torrentname and/or folder if codec is there? And if is missing in filename add it. no matter if is x264 or x265 or h264 or whatever

**Comment by fernandog** [comment]
@miigotu are you ok with that change?

**Comment by medariox** [comment]
@fernandog 
I can try that, not sure if possible tho. Anyway, I'm doing it only if it gets approved as I have no time to waste anymore.

**Comment by miigotu** [comment]
@fernandog Ill just do it another time. If it is missing in file name its just missing.

**Comment by fernandog** [comment]
but if torrentname has codec, there's no point not adding. the uploader forgot to add after re-upload the torrent/nzb

**Pull request closed without merging**